Validation 리팩토링

zlzzlzz2l opened the pull request:
# Validation 리팩토링

- Validation을 리팩토링한 후, 이메일과 닉네임 중복 체크 테스트를 작성하였습니다.

---

- 닉네임 Validation 리팩토링
- 비밀번호 Validation 추가
- 이메일 & 닉네임 중복 체크 테스트 작성
- gender, birthDate, job, role Validation 추가

zlzzlzz2l committed:
Refactor: #21 - 회원 컨트롤러 이메일 중복 체크 로직 수정

조건마다 status를 정확하게 나타내기 위해서 로직을 수정했습니다.

zlzzlzz2l committed:
Update: #21 - 회원 컨트롤러 닉네임 중복 체크 정규표현식 추가

정규표현식을 이용하여 올바르게 닉네임을 작성했는지 확인합니다.

zlzzlzz2l committed:
Refactor: #21 - 회원 컨트롤러 닉네임 중복 체크 로직 수정

조건마다 status를 정확하게 나타내기 위해서 로직을 수정했습니다.

zlzzlzz2l committed:
Update: #21 - 회원 등록요청 DTO 비밀번호 Validation 추가

zlzzlzz2l committed:
Refactor: #21 - 회원 등록요청 DTO 닉네임 Validation 수정

zlzzlzz2l committed:
Update: #21 - 회원 등록요청 DTO Builder 추가

zlzzlzz2l committed:
Create: #21 - 회원 리포지토리 테스트 생성

zlzzlzz2l committed:
Update: #21 - 회원 리포지토리 테스트 이메일 중복 체크 추가

zlzzlzz2l committed:
Update: #21 - 회원 리포지토리 테스트 닉네임 중복 체크 추가

zlzzlzz2l committed:
Create: #21 - 회원 서비스 테스트 생성

zlzzlzz2l committed:
Update: #21 - 회원 서비스 테스트 이메일 중복 체크 추가

zlzzlzz2l committed:
Update: #21 - 회원 서비스 테스트 닉네임 중복 체크 추가

zlzzlzz2l committed:
Refactor: #21 - 불필요한 패키지 제거

zlzzlzz2l committed:
Refactor: #21 - 회원 리포지토리 테스트 닉네임 중복 체크 수정

잘못 작성된 변수명을 수정하였습니다.

0xe82de reviewed (changes requested):
전반적으로 리팩토링하시고 다시 올려주세요

zlzzlzz2l committed:
Refactor: #21 - 회원 컨트롤러 닉네임 중복 체크 수정

모호한 변수명을 수정하였습니다.

zlzzlzz2l committed:
Refactor: #21 - 회원 리포지토리 테스트 이메일 중복 체크 수정

1. memberEmail을 email로 수정하였습니다.
2. 해당 이메일을 갖고 있는지 여부를 확인하기 위해 boolean 타입의 isEmail과 result를 hasEmail로 수정하였습니다.

zlzzlzz2l committed:
Refactor: #21 - 회원 리포지토리 테스트 닉네임 중복 체크 수정

1. memberNickname을 nickname으로 수정하였습니다.
2. 해당 닉네임을 갖고 있는지 여부를 확인하기 위해 boolean 타입의 isNickname과 result를 hasNickname로 수정하였습니다.

zlzzlzz2l committed:
Refactor: #21 - 회원 서비스 테스트 이메일 중복 체크 수정

해당 이메일의 중복 여부를 확인하기 위해 boolean 타입의 isEmail과 result를 isUsingEmail로 수정하였습니다.

zlzzlzz2l committed:
Refactor: #21 - 회원 서비스 테스트 닉네임 중복 체크 수정

해당 닉네임의 중복 여부를 확인하기 위해 boolean 타입의 isNickname과 result를 isUsingNickname로 수정하였습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/controller/MemberController.java:
```diff
@@ -116,13 +116,11 @@ public BaseResponseDto checkEmailDuplicate(@PathVariable String memberEmail) {
 
```
sout 다 지워 주세요 :)

zlzzlzz2l replied:
반영했습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/dto/req/ReqRegisterMemberDto.java:
```diff
@@ -13,13 +14,10 @@
-    @NotNull(message = "필수 사항입니다.")
-    @Size(min = 4, max = 10, message = "닉네임은 4자 이상 10자 이하로 입력해주세요")
-    @Pattern(regexp = "^[^ㄱ-ㅎ|ㅏ-ㅣ]*$", message = "닉네임에 자음 또는 모음을 사용할 수 없습니다.") // 자음 - 모음이 들어가면 안됨
-    @Pattern(regexp = "^[0-9|a-z|A-Z|ㄱ-ㅎ|ㅏ-ㅣ|가-힣|\\s]*$", message = "닉네임에 특수문자를 사용할 수 없습니다.") // [] 안에 들어간 문자 빼고는 안됨
+    @Pattern(regexp = "^[0-9|a-z|A-Z|ㄱ-ㅎ|ㅏ-ㅣ|가-힣|\\s]*$", message = "닉네임을 올바르게 작성해주세요.")
     private String nickname;
```
길이 제한이 적용된 것인가요?

zlzzlzz2l replied:
길이 제한이 적용된 정규표현식으로 수정하였습니다.

0xe82de commented on src/main/java/com/fourfourfour/damoa/api/member/dto/req/ReqRegisterMemberDto.java:
```diff
@@ -36,4 +34,15 @@
 
     private String role;
 
```
gender, birthDate, job, role 모두 valid 처리해주세요

zlzzlzz2l replied:
반영했습니다.

0xe82de reviewed (changes requested):
코멘트 확인해주시고 피드백 주세요 :)

yjs1213 reviewed (approved):
고생하셨습니다~

zlzzlzz2l committed:
Refactor: #21 - 회원 컨트롤러 불필요한 출력 삭제

zlzzlzz2l committed:
Refactor: #21 - 회원 등록요청 DTO 수정

1. @NoArgsConstructor 추가
2. 비밀번호 Validation 리팩토링
3. 닉네임 길이 제한 설정
4. gender, birthDate, job, role Validation 추가

0xe82de reviewed (approved):
고생하셨습니다 :)
